### PR TITLE
 Package the tool for distribution and installation 

### PR DESCRIPTION
**Pull request by siddharth-krishna**
This PR packages up our tool so that it can be distributed (eventually in the Python Package Index https://pypi.org/) and installed using `pip`.

This PR contains the following changes towards the above goal:
- Creates a `pyproject.toml` file describing the package and its dependencies (this is the modern replacement to `setup.py` files)
- Our tool can now be installed (in a virtual environment, preferably) using `pip install .`
- Moves `times_excel_reader.py` and `main.py` to `__main__.py` as is standard convention
- This allows the pip install to add a `times-excel-reader` command to the PATH, so you can invoke our tool from any directory and without needing to prefix the command with `python`!
- Config files such as `times-info.json` have been moved to `times_reader/config/` and are included in a `MANIFEST.in` file so that they are distributed along with the tool
- The GAMS scaffolding file has also been moved to `times_reader/gams_scaffold` and added to the manifest
- We now use the recommended `importlib.resource` instead of hardcoding paths to config files

One little wrinkle is that the dependencies of the tool are now listed both in `pyproject.toml` and in `requirements.txt`. We could remove the latter, but [the recommendation](https://stackoverflow.com/questions/74508024/is-requirements-txt-still-needed-when-using-pyproject-toml) is to keep both: the former lists dependencies version bounds as permissive as possible so that it doesn't conflict with other packages installed by users; the latter lists the exact dependency versions we use in development and CI so that we can reproduce bugs and issues.

Another question is the list of authors: should we list everyone who appears in https://github.com/etsap-TIMES/times-excel-reader/graphs/contributors, or ask everyone if they'd like to be included, or just leave it as "TIMES community"?

References:
- https://realpython.com/pypi-publish-python-package/
- https://packaging.python.org/en/latest/flow/
- https://drivendata.co/blog/python-packaging-2023

### CODE REVIEW

**Comment by siddharth-krishna** [comment]
I had to skip the regression tests because I've moved the `times_excel_reader.py` file. There's no (easy) way to have `run_benchmarks.py` work with the old (`python times_excel_reader.py ...`) and new (`times-excel-reader ...`) styles of calling our tool, so I'll just manually check that there are no regressions before merging. (And follow this up with a PR that turns regression testing back on)

**Comment by olejandro** [comment]
Great, thanks @siddharth-krishna! :-) Regarding authorship, let's open an issue. We should clarify it before publishing / releasing the first version, right?

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  "more-itertools",
+  "openpyxl >= 3.0, < 3.1",
+  "pandas >= 1.5",
+]
+
+[project.optional-dependencies]
+dev = ["black", "pre-commit", "tabulate"]
+
+[project.urls]
+Documentation = "https://github.com/unknown/times-excel-reader#readme"
```
Comment (olejandro) [inline pyproject.toml:34]: Should it say `etsap-times` instead of `unknown`?

**Review by olejandro — APPROVED** [review]

**Comment by siddharth-krishna** [comment]
Yep, we just need to figure out the authors list before publishing the package on the online pip repository PyPI. I'll open an issue. Thanks for the catch, I've fixed the URLs.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+[project.optional-dependencies]
+dev = ["black", "pre-commit", "tabulate"]
+
+[project.urls]
+Documentation = "https://github.com/etsap-TIMES/times-excel-reader#readme"
+Issues = "https://github.com/etsap-TIMES/times-excel-reader/issues"
+Source = "https://github.com/etsap-TIMES/times-excel-reader"
+
+[project.scripts]
+times-excel-reader = "times_reader.__main__:main"
```
Comment (siddharth-krishna) [inline pyproject.toml:35]: @olejandro this is why what we were trying today didn't work -- the entry point for the `times-excel-reader` script/command is the `main` function, so modifying the `if __name__ == "__main__"` block had no effect!

**Pull request merged**